src/components/PersonLink.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  Link,
+  useNavigate,
+  useParams,
+  useSearchParams,
+} from 'react-router-dom';
+import { Person } from '../types';
+import { useEffect, useRef } from 'react';
+
+type Props = {
+  person: Person;
+  people: Person[];
+};
+
+export const PersonLink: React.FC<Props> = ({ person, people }) => {
+  const { slug } = useParams();
+  const personRef = useRef(null);
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+
+  const handleClick = () => {
+    navigate(`/people/${person.slug}?${searchParams.toString()}`);
+  };
+
+  const findPersonByName = (name: string | null) => {
+    return people.find(p => p.name === name);
+  };
+
+  const mother = findPersonByName(person.motherName);
+  const father = findPersonByName(person.fatherName);
+
+  useEffect(() => {
+    if (slug === person.slug && personRef?.current) {
+      const observer = new IntersectionObserver(
+        ([entry]) => {
+          // eslint-disable-next-line no-console
+          console.log('entry.isIntersecting', entry.isIntersecting);
+        },
+        {
+          root: null, // viewport
+          rootMargin: '0px', // no margin
+          threshold: 0.5, // 50% of target visible
+        },
+      );
+
+      if (personRef.current) {
+        observer.observe(personRef.current);
+      }
+
+      personRef?.current.scrollIntoView({
+        behavior: 'smooth',
+        block: 'center',
+      });
+    }
+  }, [slug, personRef, person.slug]);
```

Comment:
The `useEffect` hook sets up an `IntersectionObserver` to log when the person row is in view. Ensure that the observer is disconnected when the component unmounts to prevent memory leaks. You can do this by returning a cleanup function from the `useEffect`.